components/ai_chat/resources/untrusted_conversation_frame/components/assistant_task/assistant_task.tsx

```diff
@@ -4,13 +4,178 @@
 // You can obtain one at https://mozilla.org/MPL/2.0/.
 
 import * as React from 'react'
+import Icon from '@brave/leo/react/icon'
+import ProgressRing from '@brave/leo/react/progressRing'
+import Tabs from '@brave/leo/react/tabs'
+import TabItem from '@brave/leo/react/tabItem'
+import classnames from '$web-common/classnames'
+import { getLocale } from '$web-common/locale'
 import * as Mojom from '../../../common/mojom'
+import AssistantResponse from '../assistant_response'
+import ToolEvent from '../assistant_response/tool_event'
+import styles from './assistant_task.module.scss'
+import useExtractTaskData, { TaskData } from './use_extract_task_data'
 
 interface Props {
   // Entries that make up the task loop
   assistantEntries: Mojom.ConversationTurn[]
+
+  // Whether the task can be interacted with (usually the most recent entry in
+  // the conversation). Informs whether interactivity is allowed.
+  isActiveTask: boolean
+
+  // Whether the task is actively still being generated. Informs whether the UI
+  // displays as if it's still in progress.
+  isGenerating: boolean
+
+  // Passes to AssistantResponse
+  isLeoModel: boolean
 }
 
+/**
+ * The Task is split *not* by ConversationTurn but by CompletionEvent, which is
+ * better for descriptive UI.
+ * The Progress tab displays the most recent item in that split (the
+ * completion text and the subsequent tool uses) as well as any previous
+ * "important" tool use (e.g. TODO list or tab navigation).
+ * The Steps tab displays everything in timeline order.
+ */
 export default function AssistantTask(props: Props) {
-  return <h1>Task stub</h1>
+  const [showSteps, setShowSteps] = React.useState(false)
+
+  const taskData = useExtractTaskData(props.assistantEntries)
+
+  return (
+    <div className={styles.task}>
+      <Tabs
```

Comment:
Spacing is not quite right for inside the Tabs component and should probably be solved in `@brave/leo/Tabs`:
- https://github.com/brave/leo/issues/1238